### PR TITLE
Fix/duplicate ids

### LINKED ISSUE #710: Compile subgraphERROR TS2718: Duplicate property 'id'
First time I compiled my ProxyAdminUpgradeable.
Then I tried with the implementation address and got this error.

Seems to be complaining about a BN
```
 Compile subgraphERROR TS2718: Duplicate property 'id'.

   get id(): BigInt {
       ~~
 in generated/schema.ts(54,6)

ERROR TS2718: Duplicate property 'id'.

   set id(value: BigInt) {
       ~~
 in generated/schema.ts(59,6)


ERROR AS200: Conversion from type '~lib/@graphprotocol/graph-ts/index/BigInt' to '~lib/string/String' requires an explicit cast.

   entity.id = event.params.id
               ~~~~~~~~~~~~~~~
 in src/mapping.ts(26,14)

✖ Failed to compile subgraph: Failed to compile data source mapping: Compile error
Error: Failed to compile data source mapping: Compile error
    at Compiler._compileDataSourceMapping (/home/ian/code/graph/comp/node_modules/@graphprotocol/graph-cli/src/compiler.js:314:13)
    at /home/ian/code/graph/comp/node_modules/@graphprotocol/graph-cli/src/compiler.js:207:20
    at updateInDeepMap (/home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:1971:22)
    at updateInDeepMap (/home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:1980:23)
    at updateInDeepMap (/home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:1980:23)
    at Map.updateIn (/home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:1278:26)
    at /home/ian/code/graph/comp/node_modules/@graphprotocol/graph-cli/src/compiler.js:206:24
    at /home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:3016:46
    at List.__iterate (/home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:2206:13)
    at IndexedIterable.mappedSequence.__iterateUncached (/home/ian/code/graph/comp/node_modules/immutable/dist/immutable.js:3015:23)

```

### PR DESCRIPTION
**Pull request by juanigallo**
## Description

When indexing contract events as entities, the CLI does not check if there is an "id" event. If so, the scaffolding that is generated does not compile due `Duplicate property 'id'`.

In this PR that error is fixed renaming ID entity and mapping to `{contractName}_id`

## Issues that are being fixed

Fixes #710 , #382 

## Steps to reproduce original error

If you want to test the PR solution or just take a look into the original issue, you can create a basic subgraph using the [ENS token contract](https://etherscan.io/token/0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85)

### CODE REVIEW

**Review by saihaj — CHANGES_REQUESTED** [review]
can you remove the `package-lock` files? this repo uses `pnpm` aslo wondering 🤔  if we can have a test case of this part os other CLI tests we have

**Comment by saihaj** [comment]
can you aslo add a changeset `pnpm changeset` explaining your changes learn more https://github.com/graphprotocol/graph-tooling/pull/1050#issuecomment-1407298391

File: packages/cli/src/scaffold/schema.ts [hunk 1]
```diff
@@ -50,21 +54,37 @@ export const generateEventFields = ({
...
-      transactionHash: Bytes!
-    }`;
+      
+const renameInput = (name: string, subgraphName: string) => {
+  const inputMap: BlacklistDictionary = {
+    id: `${subgraphName}_id`
+  } 
+
+  return inputMap[name] ?? name;
+}
```
Comment (saihaj) [inline packages/cli/src/scaffold/schema.ts:64]: for anyone upgrading we would need to note this is potentially breaking change

Comment (juanigallo) [reply packages/cli/src/scaffold/schema.ts:64]: Yes it is. Is there any convention or anything you use for this types of changes?

**Comment by juanigallo** [comment]
Hi @saihaj ! Just removed package-lock and added a test to check that build works as expected with the new code

**Comment by saihaj** [comment]
@juanigallo might need to run `pnpm prettier . --write` so lint doesn't fail

**Comment by juanigallo** [comment]
@saihaj is the changeset bot new in this repo? Running `pnpm changeset` throws me this error:

```
🦋  error There is no .changeset folder.
🦋  error If this is the first time `changesets` have been used in this project, run `yarn changeset init` to get set up.
🦋  error If you expected there to be changesets, you should check git history for when the folder was removed to ensure you do not lose any configuration.
```

Also running the changesets in the ["traditional way"](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) it creates a .changeset folder that appear to be different as how you are managing the changelog

**Comment by saihaj** [comment]
> @saihaj is the changeset bot new in this repo? Running `pnpm changeset` throws me this error:
> 
> ```
> 🦋  error There is no .changeset folder.
> 🦋  error If this is the first time `changesets` have been used in this project, run `yarn changeset init` to get set up.
> 🦋  error If you expected there to be changesets, you should check git history for when the folder was removed to ensure you do not lose any configuration.
> ```
> 
> Also running the changesets in the ["traditional way"](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) it creates a .changeset folder that appear to be different as how you are managing the changelog

I haven't had this issue. I use the `pnpm changeset` command every time I am publishing now 🤔

**Comment by richardryangarcia** [comment]
I pulled and ran the linting step and got no errors (only got warnings). So the linting step might pass now if someone wanted to retrigger the pipeline

**Comment by saihaj** [comment]
this branch has conflicts when trying to `git merge main`. I would recommend rebasing this and then continuing.

**Comment by saihaj** [comment]
so I am trying to push to this branch and doesn't look like I can 🤔 so will create a PR from this branch so I can fix things and land this

**Pull request closed without merging**